torch/_dynamo/variables/functions.py

```diff
@@ -1846,7 +1874,14 @@ def check_grid(self, grid) -> tuple[torch.fx.proxy.Proxy, ...]:
         if isinstance(grid, BaseListVariable):
             return grid.as_proxy()
         else:
-            unimplemented(f"grid for the triton kernel is {type(grid)}")
+            unimplemented_v2(
+                gb_type="unsupported grid type for triton hop check_grid",
+                context=f"grid type = {type(grid)}",
+                explanation="`torch.compile` currently can't trace this",
```

Comment:
Seems like we expect the input to be list-like.

Reply:
Yep updated.